ThunderBasics/ToastView.swift

```diff
@@ -91,41 +111,84 @@ public class ToastView: UIView {
     /// - Parameter completion: A completion block to be called when the toast notification has displayed and dismissed successfully
     internal func show(completion: @escaping () -> Void) {
         
-        frame = CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 44) // Add on 40 points so when it drops down you can't see the view behind it.
-        layout()
-        
-        let containerView = UIView(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: bounds.height))
+        frame = CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 44)
+                
+        let containerView = UIView(frame: .zero)
         
-        transform = CGAffineTransform(translationX: 0, y: -frame.height)
-        
-        coverWindow = UIWindow(frame: bounds)
+        coverWindow = UIWindow(frame: .zero)
         coverWindow?.isHidden = false
         coverWindow?.windowLevel = UIWindow.Level.statusBar + 1
         
         let toastViewController = ToastViewController()
         toastViewController.statusBarStyle = window?.visibleViewController?.preferredStatusBarStyle ?? .default
         
         coverWindow?.rootViewController = toastViewController
-        coverWindow?.backgroundColor = .clear
         coverWindow?.rootViewController?.view.addSubview(containerView)
         containerView.addSubview(self)
         
+        layout()
+        
+        // For some reason margins is only available at this point!
+        var safeAreaInsets: UIEdgeInsets = .zero
+        if #available(iOS 11.0, *) {
+            safeAreaInsets = UIApplication.shared.keyWindow?.rootViewController?.view.safeAreaInsets ?? .zero
+            // If safe area insets only apply to `.top` we can ignore them as we're not on a notched device and the
+            // toast already displays ABOVE the status bar so we don't need to worry about that, unless we have non-zero top margin
+            if margins.top <= 0 && safeAreaInsets.bottom == 0 && safeAreaInsets.left == 0 && safeAreaInsets.right == 0 {
+                safeAreaInsets = .zero
+            }
+        }
+                
+        let safeArea = screenPosition == .top ? safeAreaInsets.top : safeAreaInsets.bottom
```

Comment:
Not an issue for now, but if you add a `ScreenPosition` in the future it will default to below.
Preference to add a `{ get }` property to `ScreenPosition` to determine this for a the given `safeAreaInsets`.
But happy to approve without this